### ISSUE TITLE
ImportError: _align.so: undefined symbol: PyFPE_jbuf

### ISSUE THREAD

**Issue by SusanaFreitas** [marcelm/cutadapt#154]
Hello!
I am trying to use cutadapt but I still haven't managed to work around this error. As soon as I try to run the script, it stops right in the beginning with the error message:

ImportError: /home/bo1sdf/softwares/cutadapt-1.9/cutadapt/_align.so: undefined symbol: PyFPE_jbuf

I couldn't find anywhere this error... Do you think I might be doing something wrong?

Thanks in advance
Susana

**Comment by marcelm** [comment]
Was there no other message above that error? I tried to make cutadapt catch this error and print out exactly what you need to do ... Here’s the message that *should* have been printed:

> ERROR: A required extension module could not be imported because it is
> incompatible with your system. A quick fix is to recompile the extension
> modules with the following command:
> 
>    python setup.py build_ext -i

In summary, it seems you are trying to run cutadapt without compiling/installing it. This is ok and often works because I included the necessary pre-compiled modules. If your system is sufficiently different from mine, however, this will result in errors like the one you see and you need to do a normal installation. Just follow the [installation instructions](http://cutadapt.readthedocs.org/en/stable/installation.html) (or do what it says above).

**Comment by SusanaFreitas** [comment]
Hello!
Thanks for the reply. I managed to work out what was the problem. I was
using seq.fastq.gz files, and I think it was it. I read in the instructions
cutadapt could read fastq.gz files, so I thought it wouldnt be a problem.
Only after I realised I had missed the rest of the paragraph
"Note that the all of cutadapt’s options that expect a file name support
this."
When I decompressed the fastq files, it started running smoothly.
I am sorry for not letting you know immediately, I wanted to make sure I
had solved the problem first.

Thanks again!

Cheers,
Susana

2015-11-12 20:14 GMT+00:00 Marcel Martin <notifications@github.com>:

**Comment by marcelm** [comment]
It’s great you got it to work, but cutadapt *does* support compressed files, as even the sentence that you quoted says (or was at least to be supposed to). For the benefit of others who might see this: Compressed input/output files are *not* the problem, don’t uncompress your reads unnecessarily! The pasted error message is actually printed before any files are even loaded, this cannot be the problem. I believe you have changed something else that makes it working now.

**Comment by SusanaFreitas** [comment]
Dear Marcel, I am sorry about the confusion.
I misread the instructions, in fact it says it *can *use .gz files. I did
some tests and I finally found the error. When putting the file extension
.gz in the input and output files in the command line, I also added it in
the --discard-trimmed option (--discard-trimmed.gz) by mistake. So, when
running I guess it could not find this option and showed that error...

Again, I am sorry for the mistake and thanks for the support.

Best,
Susana

2015-11-12 22:11 GMT+00:00 Marcel Martin <notifications@github.com>:

**Comment by denees19** [comment]
Good afternoon, there are files in the format _Trimmed.gz (data processing after sequencing micro-RNA). The program in which it is necessary to do the analysis works only with files in the fastq format. Is it possible to do something with _Trimmed.gz to convert them to fastq?

**Comment by marcelm** [comment]
Hi, it would be more appropriate to open a new issue. Please do so next time (or send me an e-mail, that is also fine).

Was the `_Trimmed.gz` file created by Cutadapt? Then it is probably already in FASTQ format, but compressed. Perhaps you can just rename it to `_Trimmed.fastq.gz` and it may then work. If not, you could try uncompressing the file with `gunzip`.

**Comment by denees19** [comment]
Good afternoon, unfortunately, renaming the file and unpacking it did not give the desired result. If I send you a file, you can tell me what can be done with it) 19.07.2019, 16:34, "Marcel Martin" <notifications@github.com>:Hi, it would be more appropriate to open a new issue. Please do so next time (or send me an e-mail, that is also fine).Was the _Trimmed.gz file created by Cutadapt? Then it is probably already in FASTQ format, but compressed. Perhaps you can just rename it to _Trimmed.fastq.gz and it may then work. If not, you could try uncompressing the file with gunzip.—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by marcelm** [comment]
You can send the file, but I cannot promise I will be able to help.

**Comment by denees19** [comment]
Ok here's the file 24.07.2019, 12:05, "Marcel Martin" <notifications@github.com>:You can send the file, but I cannot promise I will be able to help.—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or mute the thread.   
2013.Spring.CRC.#14.Cancer.Trimmed.gz (40414303)

**Comment by marcelm** [comment]
You’ll have to send it to my proper e-mail address, not the notifications@github.com address. You’ll find the address in the documentation or run `cutadapt --help|head -n 5`.
